### ISSUE TITLE
Unclear how to map validate results to metrics when using dict

### ISSUE THREAD

**Issue by vedantroy** [mosaicml/composer#1386]
Warning:

> home/user/miniconda/envs/working/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: The ``compute`` method of metric MeanMetric was called before the ``update`` method which may lead to errors, as metric states have not yet been updated.
  warnings.warn(*args, **kwargs)

Is this is a bug? My metrics are printing `NaN` to wandb, so I suspect it might be. However, I can't tell since I can't log/raise exceptions inside of the `validate` method.

Here's my metrics collection code:

```
    def metrics(self, train: bool = False):
        assert not train, f"Metrics are not available for training"
        return torchmetrics.MetricCollection(
            {
                "mse": self.val_mse_loss,
                "vb": self.val_vb_loss,
            }
        )

    def validate(self, batch):
        out = self.forward(batch)
        losses = self.diffusion.losses_validation(
            model_output=out.model_out,
            noise=out.noise,
            x_0=out.x_0,
            x_t=out.x_t,
            t=out.t,
        )
        return { "mse": losses.mse, "vb": losses.vb }
        # return losses.mse, losses.vb
```

I need to use the dictionary form of `MetricCollection` due to this: https://github.com/mosaicml/composer/issues/1382
Also I can't seem to log stuff inside of the `validate` method or raise an Exception.

**Comment by vedantroy** [comment]
The docs say:
> We require that the output of [ComposerModel.validate()](https://docs.mosaicml.com/en/v0.8.2/api_reference/composer.models.base.html#composer.models.base.ComposerModel.validate) be consumable by torchmetrics.

I do not know what torchmetric's interface is.

**Comment by vedantroy** [comment]
I think the real find here is that I cannot log or raise an exception inside of the `validate` method which makes debugging it basically impossible. I can't put a `print` statement or `raise Exception(...)`.

**Comment by hanlint** [comment]
Hi @vedantroy , good point, we should explain the interface via the docs (filed an issue internally). the model has access to the `logger`, so you could do `model.logger.*` to log some data. Could you explain the issue you are running into for printing or raising an exception? the `validate` block above should allow for those, what error are you hitting?

**Comment by vedantroy** [comment]
@hanlint , I tried raising an exception inside the validate block and it nothing was being thrown. The program would just keep on going.

The confusing thing was, it looks like the trainer was running the evals as evidenced by my stdout output:

```

eval           Batch      6:  |                   
eval           Batch     12:  |                   
eval           Batch     18:  |                   
eval           Batch     24:  |                   
eval           Batch     30:  |                   
```

**Comment by vedantroy** [comment]
Oh, I see the issue; my val dataloader is empty & thus nothing was getting logged.
Would be nice if y'all printed a warning message if the validation data loader is empty.
@hanlint
